lib/snitcher/api/client.rb

```diff
@@ -0,0 +1,373 @@
+require "pp"
+
+require "net/https"
+require "timeout"
+require "base64"
+require "json"
+
+require "snitcher/api"
+require "snitcher/api/base"
+require "snitcher/version"
+
+class Snitcher::API::Client < Snitcher::API::Base
+  # Public: Create a new Client
+  #
+  # options:
+  #   api_key: access key available at https://deadmanssnitch.com/users/edit
+  #
+  # Example
+  #
+  #   Get the api_key for user with api_key "abc123"
+  #     @client = Snitcher::API::Client.new({api_key: "abc123"})
+  #     => #<Snitcher::API::Client:0x007fa3750af418 @api_key=abc123,
+  #          @api_endpoint=#<URI::HTTPS https://api.deadmanssnitch.com/v1/>>
+  #
+  def initialize(options = {})
+    @api_key      = options[:api_key]
+
+    ## Use in production
+    # @api_endpoint = URI.parse("https://api.deadmanssnitch.com/v1/")
+    ## Use in development for testing
+    @api_endpoint = URI.parse("http://api.dms.dev:3000/v1/")
```

Comment:
A couple different versions of the endpoint have snuck in here. Lets make it a configurable option with a default the points to production.